playground/src/editor.tsx

```diff
@@ -88,18 +90,31 @@ export function Editor(props: {
     setErrors(errList);
   }
 
-  checkRef.current = async function onCheck() {
+  checkRef.current = async function updateCode() {
+    // This should get called on initial load and on every document update.
     const code = editor.current?.getValue();
-    if (code == null) return;
-    const diags = await props.compiler.checkCode(code);
+    if (code == null) throw new Error("Why is code null?");
+
     if (props.activeTab === "hir-tab") {
       props.setHir(await props.compiler.getHir(code));
     }
-    errMarks.current.checkDiags = diags;
-    markErrors();
-    setHasCheckErrors(diags.length > 0);
+
+    const model = editor.current?.getModel();
```

Comment:
Have an 'await' call (line 99) between the calls to editor.current.getValue (for the code) and editor.current.getModel, so these could potentially be out of sync by the time you pass both to the language service. You should probably fetch everything you need from the editor within the same 'block' between await calls.

Reply:
Thanks. I actually did notice the LS was getting out of sync with keystrokes and had a fix locally, which I forgot about.

Reply:
Made some changes to how I listen to this event.